src/main/common/sdft.c

```diff
@@ -94,7 +94,7 @@ FAST_CODE void sdftPushBatch(sdft_t* sdft, const float sample, const int batchId
         batchEnd += sdft->batchSize;
     }
 
-    for (int i = batchStart; i < batchEnd; i++) {
+    for (int i = batchStart; i <= batchEnd; i++) {
```

Comment:
Extending the range like this will overlap the last of the batch with the first bin of the next batch. One bin will get processed twice and your frequency spectrum will be wrong.